internal/actions/release.go

```diff
@@ -25,30 +26,64 @@ func Release() error {
 		return err
 	}
 
-	files, err := g.GetCommitedFiles()
-	if err != nil {
-		fmt.Printf("Failed to get commited files: %s\n", err.Error())
+	dir := "."
+	usingReleasesMd := false
+	// TODO: Maybe instead check for workflow dispatch event
+	providesExplicitTarget := environment.SpecifiedTarget() != ""
+	if providesExplicitTarget {
+		workflow, err := configuration.GetWorkflowAndValidateLanguages(true)
+		if err != nil {
+			return err
+		}
+		if target, ok := workflow.Targets[environment.SpecifiedTarget()]; ok && target.Output != nil {
+			dir = strings.TrimPrefix(*target.Output, "./")
+		}
+		dir = filepath.Join(environment.GetWorkingDirectory(), dir)
 	}
 
-	if environment.IsDebugMode() {
-		for _, file := range files {
-			logging.Debug("Found commited file: %s", file)
+	if !providesExplicitTarget {
+		// This searches for files that would be referenced in the GH Action trigger
+		files, err := g.GetCommitedFiles()
+		if err != nil {
+			fmt.Printf("Failed to get commited files: %s\n", err.Error())
 		}
-	}
 
-	dir := "."
+		if environment.IsDebugMode() {
+			for _, file := range files {
+				logging.Debug("Found commited file: %s", file)
+			}
+		}
+
+		for _, file := range files {
+			// Maintain Support for RELEASES.MD for backward compatibility with existing publishing actions
+			if strings.Contains(file, "RELEASES.md") {
+				dir = filepath.Dir(file)
+				logging.Info("Found RELEASES.md in %s\n", dir)
+				usingReleasesMd = true
+				break
+			}
 
-	for _, file := range files {
-		if strings.Contains(file, "RELEASES.md") {
-			dir = filepath.Dir(file)
-			logging.Info("Found RELEASES.md in %s\n", dir)
-			break
+			if strings.Contains(file, "gen.lock") {
+				dir = filepath.Dir(file)
+				dir = strings.ReplaceAll(dir, "/.speakeasy", "")
+				dir = strings.ReplaceAll(dir, ".speakeasy", "")
```

Comment:
I don't fully understand why we're dropping the .speakeasy part of the directory

Reply:
Because we want the path to the actual SDK